### PR TITLE
fix: port changes from 0.4

### PR DESCRIPTION
**Pull request by sravankorumilli**
* default window version to 1 if not provided
* fix bug to connect to upstream optimus server for get the status of upstream jobs.
* change image pull policy to ifnotpresent

### CODE REVIEW

File: api/third_party/openapi/odpf/metadata/optimus/Job.swagger.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (sbchaos) [inline api/third_party/openapi/odpf/metadata/optimus/Job.swagger.json:1]: This file should not be part of changes. It is deleted on this branch.
Currently swagger schema is under protos.

File: api/third_party/openapi/odpf/optimus/plugins/v1beta1/dependency_resolver.swagger.json [hunk 2]
```diff
@@ -0,0 +1,156 @@
+{
```
Comment (sbchaos) [inline api/third_party/openapi/odpf/optimus/plugins/v1beta1/dependency_resolver.swagger.json:1]: Until here, all files should not be added.

File: core/job/handler/v1beta1/job_test.go [hunk 3]
```diff
@@ -547,7 +547,7 @@ func TestNewJobHandler(t *testing.T) {
 		})
 	})
 	t.Run("GetWindow", func(t *testing.T) {
-		t.Run("returns error if sheduleAt is not valid", func(t *testing.T) {
+		t.Run("returns error if sheduledAt is not valid", func(t *testing.T) {
```
Comment (sbchaos) [inline core/job/handler/v1beta1/job_test.go:550]: sheduledAt -> scheduledAt

File: api/third_party/openapi/odpf/metadata/optimus/Job.swagger.json (hunk 1 above)
Comment (sravankorumilli) [reply api/third_party/openapi/odpf/metadata/optimus/Job.swagger.json]: done

File: api/third_party/openapi/odpf/optimus/plugins/v1beta1/dependency_resolver.swagger.json (hunk 2 above)
Comment (sravankorumilli) [reply api/third_party/openapi/odpf/optimus/plugins/v1beta1/dependency_resolver.swagger.json]: done

File: core/job/handler/v1beta1/job_test.go (hunk 3 above)
Comment (sravankorumilli) [reply core/job/handler/v1beta1/job_test.go]: done

**Pull request merged**